modules/Crowd Assessment/crowdAssess_view_discuss_postProcess.php

```diff
@@ -123,16 +123,22 @@
 
                         $homeworkNameSingular = $container->get(SettingGateway::class)->getSettingByScope('Planner', 'homeworkNameSingular');
 
+                        $notificationGateway = new \Gibbon\Domain\System\NotificationGateway($pdo);
+
                         //Create notification for homework owner, as long as it is not me.
                         if ($gibbonPersonID != $session->get('gibbonPersonID') and $gibbonPersonID != $replyToID) {
                             $notificationText = __('Someone has commented on your {homeworkName} for lesson plan "{lessonName}".', ['lessonName' => $name, 'homeworkName' => mb_strtolower(__($homeworkNameSingular))]);
-                            setNotification($connection2, $guid, $gibbonPersonID, $notificationText, 'Crowd Assessment', "/index.php?q=/modules/Crowd Assessment/crowdAssess_view_discuss.php&gibbonPlannerEntryID=$gibbonPlannerEntryID&gibbonPlannerEntryHomeworkID=$gibbonPlannerEntryHomeworkID&gibbonPersonID=$gibbonPersonID");
+                            $notificationSender = new \Gibbon\Comms\NotificationSender($notificationGateway, $session);
```

Comment:
A more DRY way to write this could be to initialize a single sender below the gateway, then add notifications as needed in your conditionals, and run the `$notificationSender->sendNotifications();` method once at the bottom of this scope. The sender will know how to handle cases where there are no notifications to send.

Reply:
Thanks for this. It is good to know how it works, even keeping track of the separate text notifications. I've committed all the suggested changes.